File: examples/dart/async_task.dart
import 'dart:io';

import 'package:ailabtools/ailabtools.dart';

Future<void> main() async {
  final client = AILabClient(apiKey: Platform.environment['AILAB_API_KEY']!);
  try {
    final bytes = await File('image.jpg').readAsBytes();
    final task = await client.portrait.changeHairstyle(
      PortraitHairstyleEditingProParams(
        taskType: 'async',
        image: AILabFile.fromBytes(bytes, filename: 'image.jpg'),
        hairStyle: 'BuzzCut',
      ),
    );
    final taskId = task.taskId ?? task.data?.taskId;
    final result = await client.waitForTask(taskId!);
    print(result.data);
  } finally {
    client.close();
  }
}

File: examples/php/async_task.php
<?php

declare(strict_types=1);

require __DIR__ . '/../../packages/php/vendor/autoload.php';

use AILabTools\AILabClient;
use AILabTools\Generated\Params\PortraitHairstyleEditingProParams;
use AILabTools\Http\FileInput;

$client = new AILabClient(apiKey: getenv('AILAB_API_KEY') ?: '');
$task = $client->portrait->changeHairstyle(
    new PortraitHairstyleEditingProParams(
        taskType: 'async',
        image: FileInput::fromPath(__DIR__ . '/../../image.jpg'),
        hairStyle: 'BuzzCut',
    ),
);

$taskId = $task->taskId ?? $task->data?->taskId;
$result = $client->waitForTask(taskId: $taskId);
var_dump($result->data);
